packages/regl-worldview/src/Worldview.js

```diff
@@ -11,49 +11,40 @@ import pickBy from "lodash/pickBy";
 import * as React from "react";
 import ContainerDimensions from "react-container-dimensions";
 
-import { CameraListener } from "./camera/index";
+import { CameraListener, DEFAULT_CAMERA_STATE } from "./camera/index";
 import type { MouseHandler, Dimensions, Vec4, CameraState } from "./types";
 import { Ray } from "./utils/Raycast";
 import { WorldviewContext } from "./WorldviewContext";
 import WorldviewReactContext from "./WorldviewReactContext";
 
 const DEFAULT_BACKGROUND_COLOR = [0, 0, 0, 1];
 
-type SharedProps = {|
+export type BaseProps = {|
   backgroundColor?: Vec4,
   hitmapOnMouseMove?: boolean,
-  hideDebug?: boolean,
+  showDebug?: boolean,
   children?: React.Node,
 
+  defaultCameraState: CameraState,
```

Comment:
I thought we intentionally kept them separate in the past to prevent people from passing _both_ defaultCameraState and cameraState?

Reply:
with the api change, thought it's no longer necessary to keep it so strict. There are a few scenarios:
- nothing is passed in => uncontrolled, DEFAULT_CAMERA_STATE will apply
- defaultState => uncontrolled 
- state => uncontrolled, state is respected 
- state, defaultState => uncontrolled, defaultState will get ignored 
- onChange => uncontrolled, DEFAULT_CAMERA_STATE will apply
- defaultState, onChange => uncontrolled

- state, onChange => controlled 
- state, defaultState, onChange => controlled, defaultState will get ignored 

The behavior is similar to text input.

Reply:
Hm I think these are not quite right:

> state => uncontrolled, state is respected
> state, defaultState => uncontrolled, defaultState will get ignored

Should both be controlled; read-only.

Reply:
> defaultState will get ignored

Isn't this kind of a bug? On the other hand, maybe it makes things easier for some use cases if we allow this. I don't feel strongly.

Reply:
I've updated in https://github.com/cruise-automation/webviz/pull/40/commits/6c453ccf90fd8e77f56d7f21ef93b3ddd73b6d9a. I think it's more accurate to not default the defaultState in static defaultProps, but in constructor, so that we could provide more explicit warnings.